docs/tutorial/frontend-tut.md

```diff
@@ -99,42 +122,43 @@ component main() -> {
 
 The [cells section][lf-cells] instantiates hardware units like adders, memories and registers.
 The [wires section][lf-wires] contains [groups][lf-groups] that connect
-together hardware instances to perform some logical task such as incrementing a specific register.
+hardware instances to perform some logical task (e.g, incrementing a register).
 Finally, the [control section][lf-control] *schedules* the execution of groups using control operators such as `seq`, `par`, and `while`.
 
-We perform syntax-directed compilation by walking over nodes in the above AST and generating `cells`, `wires`, and `control` operations.
+We perform syntax-directed compilation by walking over the nodes of the AST and generating `cells`, `wires`, and `control` operations.
 
-#### Calyx Embedded DSL
 
-To make it easy to generate the hardware, we'll use Calyx's [`builder` module][builder-ex] in Python:
+### Calyx-Embedded DSL
+
+To make it easy to generate hardware, we'll use Calyx's [`builder` module][builder-ex] written in Python:
 ```python
 import calyx.builder as cb
 
 prog = cb.Builder() # A Calyx program
 main = prog.component("main") # Create a component named "main"
 ```
 
-#### `Decl` nodes
+
+### `Decl` nodes
 
 `Decl` nodes instantiate new memories and registers.
 We need these to be instantiated in the `cells` section of our Calyx output.
 We use Calyx's `std_reg` and `std_mem_d1` primitives to represent registers and memories:
-
 ```C
 import "primitives/core.futil"; // Import standard library
 
 component main() -> () {
   cells {
     // A memory with 4 32-bit elements. Indexed using a 6-bit value.
-    foo = std_mem_d1(32, 4, 6);
+    avec = std_mem_d1(32, 4, 6);
     // A register that contains a 32-bit value
-    r = std_reg(32);
+    sos = std_reg(32);
   }
   ...
 }
 ```
 
-For each `Decl` node, we need to determine if we're instantiating a memory or a register, and then translate that to a corresponding Calyx declaration and place that inside the `cells` section of our generated program.
+For each `Decl` node, we need to determine if we're instantiating a memory or a register, translate the node into a corresponding Calyx declaration, and place the declaration inside the `cells` section of our generated program.
```

Comment:
The part after this is ripe for rejiggering. We should show people what the `builder` code looks like and how it generates the Calyx code